components/sync15/src/telemetry.rs

```diff
@@ -630,3 +630,86 @@ mod sync_tests {
         );
     }
 }
+
+/// The Sync ping payload
+/// May have many syncs, may have many events. However, due to the architecture
+/// of apps which use these components, this payload is almost certainly not
+/// suitable for submitting directly. For example, we will always return a
+/// payload with exactly 1 sync, and it will not know certain other fields
+/// in the payload, such as the FxA device ID. The intention is that comsumers
```

Comment:
Maybe also note that it's not the direct device id, it's hashed with the uid (as in https://searchfox.org/mozilla-central/source/services/sync/modules/browserid_identity.js#185). Mainly to increase the chance that we/whoever implements that bit remembers to do that correctly.